utils.py

```diff
@@ -1,32 +1,57 @@
 import os
 import subprocess
 import time
+import warnings
 
 
-async def download_and_extract_tarball(
-    url: str,
-) -> str:
+async def resolve_model_path(url_or_local_path: str) -> str:
     """
-    Downloads a tarball from `url` and extracts to `dirname` if `dirname` does not exist.
+    Resolves the model path, downloading if necessary.
 
     Args:
-        url (str): URL to the tarball
+        url_or_local_path (str): URL to the tarball or local path to a directory containing the model artifacts.
 
     Returns:
-        path (str): Path to the directory where the tarball was extracted
+        str: Path to the directory containing the model artifacts.
     """
+    if "://" in url_or_local_path:
+        return await download_tarball(url_or_local_path)
+    else:
```

Comment:
could we use `file://`?

Reply:
Good callout. For completeness, I guess we should support `file:` scheme but not require it.

FYI - The File URI scheme is tricky because clients just as often use 0, 1, 2, or 3 leading slashes, so be careful resolving that.

Reply:
@joehoover Pushed ac08589ca7f39a5f56fd5f94c339b63ce11d61ec making this change. Feel free to back out of that if you think that's overkill.